### DISCUSSION TITLE
Can you tell about the support of the no_proxy ENV variable?

### DISCUSSION THREAD

**Discussion by nonoxp** [Azure/azure-storage-fuse#1077] [General]
Hi,
I have an Ubuntu environment (VM in a VNET) that requires the use of a proxy per company policy.
Unfortunately said proxy is not dealing well with access to our Storage accounts, and this proxy cannot be changed.

In order to access an Azure Storage account (private link) from this environment, we setup the no_proxy variable and its uppercase version to exclude all Azure storage blob endpoints (e.g.: export no_proxy=.blob.core.windows.net,.dfs.core.windows.net).

This works for any other tool on the pc, when I curl into the azure storage I have no problem getting a file from there as the proxy is skipped for this domain.
But this does not work when using the blobfuse2 tool.
I checked it works if I remove the http_proxy/https_proxy environment variables (but that is against policy).

Do you know if the usage of the no_proxy environment variable is taken in account?
It should be since it recognizes the http_proxy environment variable, although Go favors the uppercase NO_PROXY environment variable.
I tried to force the http-proxy and https-proxy settings to null or empty string in the setting file, did not work.
I did not find anything in the code related to this behavior, just wondering what may be the cause...

**Comment by bhansalivikas** [comment]
If http_proxy is set, then Blobfuse will use them. To avoid using them you need to reset the env variables. In past we have also seen there exists some config files which override the current env settings and update the proxy settings, so you need to be careful about those.
